lib/Ref/Util/Rewriter.pm

```diff
@@ -171,6 +169,11 @@ sub rewrite_doc {
             }
 
             $ref_sub->remove;
+
+            # update statements... to avoid PPI issues when moving elements...
+            # this is very ugly... but probably the best solution
+            $all_statements = $doc->find('PPI::Statement');
+            goto ALL_STATEMENTS;
```

Comment:
I don't like this either, but I couldn't think of a better way to do it at the moment. I moved it to a function with a `while` but it didn't work well. But I would really appreciate coming up with a better way to do this.